staging/src/k8s.io/kubectl/pkg/cmd/version/skew_warning.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"io"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"regexp"
+	"strconv"
+)
+
+const supportedMinorVersionSkew = 1
+
+func printVersionSkewWarning(w io.Writer, clientVersion, serverVersion apimachineryversion.Info) error {
+	if supported, err := isWithinSupportedVersionSkew(clientVersion, serverVersion); err != nil {
+		return err
+	} else if !supported {
+		fmt.Fprintf(w, "WARNING: version difference between client (%s.%s) and server (%s.%s) exceeds the supported minor version skew of +/-%d\n",
+			clientVersion.Major, clientVersion.Minor, serverVersion.Major, serverVersion.Minor, supportedMinorVersionSkew)
+	}
+	return nil
+}
+
+func isWithinSupportedVersionSkew(clientVersion, serverVersion apimachineryversion.Info) (bool, error) {
+	clientMajorVersion, clientMinorVersion, err := parseMajorMinorVersion(clientVersion)
+	if err != nil {
+		return false, err
+	}
+
+	serverMajorVersion, serverMinorVersion, err := parseMajorMinorVersion(serverVersion)
+	if err != nil {
+		return false, err
+	}
+
+	return clientMajorVersion == serverMajorVersion &&
+		clientMinorVersion >= serverMinorVersion-supportedMinorVersionSkew &&
+		clientMinorVersion <= serverMinorVersion+supportedMinorVersionSkew, nil
```

Comment:
To be honest (and this probably is because I'm pretty a serial reader...) this seemed confused to me :)

It might be worst but more "readable", maybe having a math.Abs(int(clientVersion - serverVersion)) and checking if this is bigger then the Skew :)

But I might be wrong, or maybe just bikeshedding here so feel free to ignore :)